backend/service/auditsink/slack/slack.go

```diff
@@ -104,3 +123,90 @@ func formatText(username string, event *auditv1.RequestEvent) string {
 
 	return messageText
 }
+
+// FormatCustomText parses out the audit event metdata request for the custom slack message text
+func FormatCustomText(message string, event *auditv1.RequestEvent) (string, error) {
+	tmpl, err := template.New("customText").Funcs(funcMap).Parse(message)
+	if err != nil {
+		return "", err
+	}
+
+	data, err := getAuditMetadata(event)
+	if err != nil {
+		return "", err
+	}
+
+	var buf bytes.Buffer
+	if err := tmpl.Execute(&buf, data); err != nil {
+		return "", err
+	}
+
+	return buf.String(), nil
+}
+
+// returns the API request/response details saved in an audit event
+func getAuditMetadata(event *auditv1.RequestEvent) (*auditEventMetadata, error) {
```

Comment:
changed this up to go from proto -> JSON -> map[string]interface{}. the map[string]interface{} is easier to work with than proto, thanks for the the suggestion!